client/src/components/UserProfile.vue

```diff
@@ -0,0 +1,123 @@
+<template>
+	<v-container>
+		<v-row>
+			<v-col cols="12" sm="12">
+				<v-text-field
+					v-model="formData.username"
+					class="my-2"
+					:rules="rules.username"
+					label="Username"
+					:readonly="isReadOnly"
+					dark
+					outlined
+					required
+				/>
+				<v-card-actions>
+					<v-btn v-if="!isReadOnly" @click="resetForm"> Cancel </v-btn>
+					<v-btn
+						color="secondary"
+						:disabled="!formIsValid"
+						@click="updateUserData"
+					>
+						{{ isReadOnly ? "Edit" : "Save" }}
+					</v-btn>
+				</v-card-actions>
+			</v-col>
+		</v-row>
+		<v-snackbar
```

Comment:
There is TheAlert component that could be used as a notification

Reply:
Hey @DavidGolodetsky , makes sense, replaced `VSnackbar` with `TheAlert` component.